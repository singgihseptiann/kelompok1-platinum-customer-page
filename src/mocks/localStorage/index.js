const mockLocalStorage = () => {
  const setItemMock = jest.fn();
  const getItemMock = jest.fn();
  const removeItemMock = jest.fn();

  beforeEach(() => {
    // localStorage.setItem()
    Storage.prototype.setItem = setItemMock;
    // localStorage.getItem()
    Storage.prototype.getItem = getItemMock;
    // localStorage.removeItem()
    Storage.prototype.removeItem = removeItemMock;
  });

  afterEach(() => {
    setItemMock.mockRestore();
    getItemMock.mockRestore();
    removeItemMock.mockRestore();
  });

  return { setItemMock, getItemMock, removeItemMock };
};

export { mockLocalStorage };
