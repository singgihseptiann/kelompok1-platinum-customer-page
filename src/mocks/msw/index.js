import { setupServer } from "msw/node";

import { handlers } from "../msw/handler";

const mockServer = setupServer(...handlers);

export default mockServer;
