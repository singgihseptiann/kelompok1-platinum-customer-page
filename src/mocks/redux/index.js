import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";

/**
 * import store yg sudah dibuat
 */
import { store } from "../../store";

const renderWithProviders = (ui, { preloadedState = {}, appStore = store, ...renderOptions } = {}) => {
  const wrapper = ({ children }) => {
    return (
      <BrowserRouter>
        <Provider store={appStore}>{children}</Provider>
      </BrowserRouter>
    );
  };

  return {
    appStore,
    ...render(ui, { wrapper, ...renderOptions }),
  };
};

const rawRenderWithProviders = (ui) => {
  return (
    <BrowserRouter>
      <Provider store={store}>{ui}</Provider>
    </BrowserRouter>
  );
};

export { renderWithProviders, rawRenderWithProviders };
