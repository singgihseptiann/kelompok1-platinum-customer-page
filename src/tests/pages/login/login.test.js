import renderer from "react-test-renderer";
import { describe, expect, test } from "@jest/globals";
import { fireEvent, screen, waitFor } from "@testing-library/react";

import { rawRenderWithProviders, renderWithProviders } from "../../../mocks/redux";
import { mockLocalStorage } from "../../../mocks/localStorage";
import LoginPage from "../../pages/login";

describe("LoginPage Component", () => {
  const { setItemMock } = mockLocalStorage();

  test("Login component should match with snapshot", () => {
    const component = renderer.create(rawRenderWithProviders(<LoginPage />)).toJSON();

    expect(component).toMatchSnapshot();
  });

  test("Should display error message on failed login attempt", async () => {
    renderWithProviders(<LoginPage />);

    const loginButton = screen.getByRole("button", { name: /sign in/i });

    fireEvent.click(loginButton);

    await waitFor(() => {
      expect(screen.getByText("Email or password is incorrect")).toBeVisible();
    });
  });

  test("Should navigate to home on successful login", async () => {
    renderWithProviders(<LoginPage />);

    const emailInput = screen.getByLabelText(/email/i);
    const passwordInput = screen.getByLabelText(/password/i);
    const loginButton = screen.getByRole("button", { name: /sign in/i });

    fireEvent.change(emailInput, { target: { value: "admin@bcr.io" } });
    fireEvent.change(passwordInput, { target: { value: "123456" } });
    fireEvent.click(loginButton);

    await waitFor(() => {
      expect(setItemMock).toHaveBeenCalledWith("token", "dummy-token");
      // Simulate a delay, then check for the navigation
      setTimeout(() => {
        expect(window.location.pathname).toBe("/home");
      }, 1500);
    });
  });

  // Additional test cases for form input validation, show/hide password, etc.
});
