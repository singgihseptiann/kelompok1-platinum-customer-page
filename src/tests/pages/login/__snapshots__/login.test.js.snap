// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage Component Login component should match with snapshot 1`] = `
<main>
  <section>
    <div
      className="container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="col-md-6 d-flex flex-column justify-content-center col"
        >
          <div
            className="mx-auto"
          >
            <div
              className="mb-4 "
              style={
                Object {
                  "backgroundColor": "#CFD4ED",
                  "height": "34px",
                  "width": "80px",
                }
              }
            />
            <h3
              className="mb-4"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Welcome Back!
            </h3>
            <div>
              <form
                className=""
                method="post"
                onSubmit={[Function]}
              >
                <div
                  className="mb-4"
                >
                  <label
                    className="form-label"
                    htmlFor="email"
                  >
                    Email
                  </label>
                  <input
                    className="form-control"
                    name="email"
                    onChange={[Function]}
                    placeholder="Example: johndee@gmail.com"
                    type="email"
                    value=""
                  />
                </div>
                <div
                  className="mb-4"
                >
                  <label
                    className="form-label"
                    htmlFor="password"
                  >
                    Password
                  </label>
                  <input
                    className="form-control"
                    name="password"
                    onChange={[Function]}
                    placeholder="6+ characters"
                    type="password"
                    value=""
                  />
                  <div
                    className="d-flex flex-row justify-space-between align-items-center"
                  >
                    <button
                      className="password-toggle btn btn-link"
                      disabled={false}
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        strokeWidth="0"
                        style={
                          Object {
                            "color": undefined,
                          }
                        }
                        viewBox="0 0 1024 1024"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M396 512a112 112 0 1 0 224 0 112 112 0 1 0-224 0zm546.2-25.8C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 0 0 0 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM508 688c-97.2 0-176-78.8-176-176s78.8-176 176-176 176 78.8 176 176-78.8 176-176 176z"
                        />
                      </svg>
                    </button>
                    <div>
                      Show Password
                    </div>
                  </div>
                </div>
                <button
                  className="w-100 btn btn-primary"
                  disabled={false}
                  type="submit"
                >
                  Sign In
                </button>
              </form>
            </div>
            <p
              className="text-center mt-4"
            >
              Donâ€™t have an account?
               
              <a
                href="/signup"
                onClick={[Function]}
              >
                Sign Up For Free
              </a>
            </p>
          </div>
        </div>
        <div
          className="col-md-6 d-none d-md-block col"
        >
          <img
            alt="Login"
            className="login-img img-fluid"
            src="/images/login.png"
          />
        </div>
      </div>
    </div>
  </section>
</main>
`;
